TFNetworkRecLayer.py

```diff
@@ -4501,12 +4502,23 @@ def get_sub_layer_out_data_from_opts(cls, layer_name, parent_layer_kwargs):
     targets = parent_layer_kwargs["target"]
     assert isinstance(targets, list), "Sub-layers for ChoiceLayer should only exist in case of multiple targets."
 
+    parent_layer_name = parent_layer_kwargs["name"]
+    network = parent_layer_kwargs["network"]
+
     # The sub-layer with index n will output the n-th target. The out_data is taken directly
     # from the target as it is done in self.get_out_data_from_opts().
-    sub_layer_out_data = cls.get_out_data_from_opts(target=targets[index], network=parent_layer_kwargs["network"],
+    sub_layer_out_data = cls.get_out_data_from_opts(name=parent_layer_name + "/" + layer_name,
+                                                    sources=parent_layer_kwargs["sources"],
+                                                    target=targets[index],
+                                                    network=network,
                                                     beam_size=parent_layer_kwargs["beam_size"])
+
+    if network.search_flag:
+        # Set name of beam to parent_layer, so that they compare to equal.
+        sub_layer_out_data.beam.name = "%s%s" % (network.get_absolute_name_prefix(), parent_layer_name)
```

Comment:
We indent by 2 spaces.

Reply:
Btw, regarding the code: Instead of overwriting the `beam.name`, maybe better resetting the `beam` itself, to `parent_layer.beam` or so? I feel that this is a bit ugly, and has certain assumptions hard-coded about the name of the beam, which might easily break later.

Reply:
I agree this is not clean. But `get_sub_layer_out_data_from_opts()` has no access to the parent layer (as it's a classmethod), only its arguments.
I think I will create a new `SearchBeam` object in this place using the parent layer name, this should be better.